### PR TITLE
docs: Golang function developer's guide

### PR DESCRIPTION
**Pull request by zroubalik**
Signed-off-by: Zbynek Roubalik <zroubali@redhat.com>

### CODE REVIEW

File: docs/guides/golang.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+```
+❯ go test
+```
+
+## Function reference
+
+Boson Go functions have very few restrictions. You can add any required dependencies
+in `go.mod` and you may include additional local Go files. The only real requirement are 
+that your project is defined in `function` module and exports function `Handle()`
```
Comment (lance) [inline docs/guides/golang.md:84]:
```suggestion
that your project is defined in a `function` module and exports the function `Handle()` 
```

File: docs/guides/golang.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+
+  // Process body & function logic
+  // ...
+}
+```
+
+#### Function triggered by CloudEvent
+
+If the incoming request is a `CloudEvent`, the event is provided via
+[CloudEvents Golang SDK](https://cloudevents.github.io/sdk-go/) and it's `Event` type
```
Comment (lance) [inline docs/guides/golang.md:125]:
```suggestion
[CloudEvents Golang SDK](https://cloudevents.github.io/sdk-go/) and its `Event` type
```

**Review by lance — APPROVED** [review]
Looks good - ty! Just a couple of very minor suggestions.

File: docs/guides/golang.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+}
+```
+
+Or we can use Golang's `encoding/json` package to access the `CloudEvent` directly as
+a JSON in form of bytes array:
+
+```golang
+func Handle(ctx context.Context, event cloudevents.Event) {
+
+  bytes, err := json.Marshal(event)
```
Comment (matejvasek) [inline docs/guides/golang.md:181]: I think that you should call `event.Data()` when working with raw data (e.g. if data are *.jpeg). Am I right @slinkydeveloper ?

Comment (zroubalik) [reply docs/guides/golang.md:181]: This is what [CE Go SDK ](https://cloudevents.github.io/sdk-go/#serializedeserialize-a-cloudevent) mentions for marshalling to json.

Comment (lance) [reply docs/guides/golang.md:181]: @zroubalik @matejvasek are we good to land this?

Comment (zroubalik) [reply docs/guides/golang.md:181]: I am, not sure what @matejvasek thinks

Comment (matejvasek) [reply docs/guides/golang.md:181]: I thinks this is all right.  I just don't see why I would transformed CE to JSON.

**Review by matejvasek — APPROVED** [review]

**Pull request merged**